core/main/src/firebolt/handlers/metrics_management_rpc.rs

```diff
@@ -105,7 +107,9 @@ impl MetricsManagementServer for MetricsManagementImpl {
         for key in request.keys {
             // currently handling only one key which is deviceSessionId
             if key.as_str() == "deviceSessionId" {
-                self.state.metrics.update_session_id(None);
+                self.state
+                    .metrics
+                    .update_session_id(Some(Uuid::new_v4().to_string()));
```

Comment:
As per the requirement, session_id should MetricsManagement.removeContext should revert Ripple back to its fallback value.  That value is the one created during `initialize`  not a new Uuid.